prov/sm2/src/sm2_coordination.c

```diff
@@ -62,6 +62,18 @@ static inline bool pid_lives(int pid)
 	return err == 0;
 }
 
+static inline int sm2_mmap_check_version(struct sm2_coord_file_header* tmp_header) {
+	if (tmp_header->file_version == SM2_VERSION)
+		return 0;
```

Comment:
And before you ask @a-szegel, I did actually go to re-write this function to return a ssize_t as you'd likely prefer, but I find that most of our error paths within `sm2_file_open_or_create` come from system calls which return `int`'s, so it seemed cleaner to be consistent within the function and keep `err` as an `int`, along with all the functions which might set `err`.